### DISCUSSION TITLE
Suggestion on improving the current scoring system.

### DISCUSSION THREAD

**Discussion by ArchUsr64** [ppy/osu#14757] [Ideas]
Current scoring system in lazer works really well but it still has a few flaws in my opinion.
First being that the scores don't mean much and only really useful when comparing two scores, secondly it being very combo dependent. This screenshot shows that issue quite clearly where a 17% accuracy FC score beats a 1 miss SS choke by a very wide margin(on [this](https://osu.ppy.sh/beatmapsets/1514884#osu/3102785) hitcircle only beatmap).

![problemWithScoring](https://user-images.githubusercontent.com/83179501/133370740-b452c7a9-9c1e-4844-8430-450a991754e2.jpeg)

I suggest a scoring system where the score relates to "_how close the play was to being an SS_", with a 1 million being an SS similar to how it currently works. Where a score of 750k could either refer to an FC with 75% accuracy or a 1 miss SS choke at the middle of the map(500k for SSing the first half and 250k for SSing the second half with half the combo).
This would make each three performance metrics namely rank(or accuracy), score and PP unique and useful in their own ways without the score being just a rough analogue for max combo as it currently is.

An example of current implementation vs my suggestion can be seen in the following clip created using [this](https://www.desmos.com/calculator/fzezuuebb8) desmos graph. The illustration shows comparison of the final scores obtained with each scoring methods when the player gets a 1 miss FC choke at a given time into the song with `T` overall accuracy.
(**Note**: Both the `T` accuracy and the value on x-axis depicting the length of song are fractional values not percentages)
https://user-images.githubusercontent.com/83179501/133373604-d03a65cd-a3f2-4c33-80d1-203600d76025.mp4

The way it works mathematically is by instead of using some menifestation of `currentCombo` as a multiplier, it loosely uses `1 - (abs(currentCombo - currentComboIfTheScoreWasAnSS)/maxPossibleComboOnTheMap)`. This value results in a number between 1 and 0 which roughly tells us "_how much should the rest of the map be weighed if FCed(without taking accuracy into account)_", like in a score with 1 miss SS choke at the middle of the map, the value would be 1 for the first half but 0.5 for the second giving a final result of 750k which if the accuracy was 100% would result in a final score of 750k with other accuracies linearly decreasing the score. As shown in the illustration earlier, this makes for a much less combo dependent design as long as the max combo is unimpressive relative to the maximum possible combo on the map.

Might need some tweaking with how the new "multiplier" and accuracy interface, not sure if linear scoring of scores based on accuracy alone would be fun but I think this implementation is much more intuitive and better for comparing the skill required in achieving each score with a much higher emphasis on accuracy and combo still having a meaningful impact.

**Comment by peppy** [comment]
The score system internally has a per-ruleset combo/accuracy ratio, which is currently not factored into grades, which is why you're seeing this. In osu! this is 80% combo 20% accuracy, while in osu!mania it's 99% accuracy 1% combo. This will continue to be tweaked, but doesn't seem too relevant to the issue you present.

So put a slightly different way, would you say that you'd prefer the grades to be based on score, not accuracy? If so, I can quite strongly agree with that.

**Reply by ArchUsr64** [reply]
Not really, in it's current state I don't think score based grades would be very intuitive and would encourage low accuracy high combo scores even more which I don't think is the right way of doing things in osu!.

**Reply by peppy** [reply]
If that's the case then you probably want to check #14078 to achieve a rough correlation between the two, I guess.

**Comment by gagahpangeran** [comment]
Related discussion about changing accuracy and combo ratio https://github.com/ppy/osu/discussions/14078
